### PR TITLE
Add support for isTypeof on Object type.

### PR DESCRIPTION
**Pull request by ruiaraujo**
It will validate objects before using them if defined. It will not support promises for now.

### CODE REVIEW

File: src/execution.ts [hunk 1]
```diff
@@ -604,23 +610,41 @@ function compileLeafType(
...
   fieldMap: { [key: string]: FieldNode[] },
   alwaysDefer: boolean
 ): string {
-  let body = `{`;
+  let body = "(";
+  if (typeof type.isTypeOf === "function" && !alwaysDefer) {
+    context.dependencies.set(type.name + "IsTypeOf", type.isTypeOf);
+    body += `!${type.name}IsTypeOf(${originPaths.join(
+      "."
+    )}) ? (${errorDestination}.push(${createErrorObject(
```
Comment (boopathi) [inline src/execution.ts]: this could be an if..else. what do you think ... both branches are used as statements rather than expressions.

Comment (ruiaraujo) [reply src/execution.ts]: This is inside a giant expression so I can't really change it in this PR.

**Comment by boopathi** [comment]
👍

**Comment by ruiaraujo** [comment]
:+1:

**Pull request merged**